### ISSUE TITLE
`eachAsync` doesn't seem to support callback usage

### ISSUE THREAD

**Issue by chrisfosterelli** [neumino/rethinkdbdash#251]
The [documentation on `eachAsync`](https://rethinkdb.com/api/javascript/each_async/) has the following example in it:

```javascript
// process each row asynchronously
cursor.eachAsync(function (row) {
    doSomethingWith(row);
});

// as above, but using rowFinished callback
cursor.eachAsync(function (row, rowFinished) {
    doSomethingWith(row);
    rowFinished();
});
```

I believe the idea is that, in the first example, `doSomethingWith` is called as records stream in from the database. In the second example, the stream is paused until `rowFinished` is called. Looking at the code, rethinkdbdash only seems to support use cases where promises are returned from the handling function. When I attempt to use the callback method I get:

```
TypeError: Cannot read property 'then' of undefined
```

Which I'm guessing happens [here](https://github.com/neumino/rethinkdbdash/blob/cc4d706dfb101701fffdcf55d645d662e80a770b/lib/cursor.js#L268) in the codebase.

**Comment by neumino** [comment]
I think the idea is to return a promise and that the next row won't be consume until your promise is resolved.

**Comment by chrisfosterelli** [comment]
Yeah that's my thought as well! But they appear to support flow management
with callbacks or with no flow managment at all in addition to promise flow
management.

Rethinkdbdash doesnt seem to implement the full rethinkdb cursor, just the
promise part.

This would be useful for working with libraries that are callback based
(although adding bluebird to the project and using promisify is a hacky
workaround in the meantime), or if you have code that doesn't need flow
control.

**Comment by marshall007** [comment]
@neumino I think the callback-style was added to `eachAsync` in 2.3 since `each` was being deprecated.

**Comment by neumino** [comment]
Fixed in 2.3.7.

Note that
```
cursor.eachAsync(function (row, rowFinished) {
    doSomethingWith(row);
    rowFinished();
    return Promise.resolve();
});
```

will behave asynchronously giving priority to the callback. I still think that people shouldn't use the callback, especially since bluebird let you easily promisify things.

**Comment by chrisfosterelli** [comment]
Looks great, thanks @neumino!
